src/DataTypes/Serializations/SerializationArrayOffsets.cpp

```diff
@@ -0,0 +1,70 @@
+#include <Columns/ColumnsNumber.h>
+#include <DataTypes/Serializations/SerializationArrayOffsets.h>
+
+namespace DB
+{
+
+void SerializationArrayOffsets::deserializeBinaryBulkWithMultipleStreams(
+    ColumnPtr & column,
+    size_t rows_offset,
+    size_t limit,
+    DeserializeBinaryBulkSettings & settings,
+    DeserializeBinaryBulkStatePtr &,
+    SubstreamsCache * cache) const
+{
+    settings.path.push_back(Substream::Regular);
```

Comment:
Should `Substream::Regular` be pushed for this substream? It appears that `SerializationArray::deserializeBinaryBulkWithMultipleStreams` doesn’t do this.